crates/core/src/surfnet/locker.rs

```diff
@@ -313,6 +320,87 @@ impl SurfnetSvmLocker {
     }
 }
 
+/// Get signatures for Addresses
+impl SurfnetSvmLocker {
+    pub fn get_signatures_for_address_local(
+        &self,
+        pubkey: &Pubkey,
+    ) -> SvmAccessContext<Vec<RpcConfirmedTransactionStatusWithSignature>> {
+        self.with_contextualized_svm_reader(|svm_reader| {
+            let current_slot = svm_reader.get_latest_absolute_slot();
+
+            svm_reader
+                .transactions
+                .iter()
+                .filter_map(|(sig, status)| {
+                    let TransactionWithStatusMeta(slot, tx, _, err) = status.expect_processed();
+
+                    // Check if the pubkey is a signer
+                    let is_signer = tx
+                        .message
+                        .static_account_keys()
+                        .iter()
+                        .position(|pk| pk == pubkey)
+                        .map(|i| tx.message.is_signer(i))
+                        .unwrap_or(false);
+
+                    if !is_signer {
+                        return None;
+                    }
+
+                    // Determine confirmation status
+                    let confirmation_status = match current_slot {
+                        cs if cs == *slot => SolanaTransactionConfirmationStatus::Processed,
+                        cs if cs < slot + FINALIZATION_SLOT_THRESHOLD => {
+                            SolanaTransactionConfirmationStatus::Confirmed
+                        }
+                        _ => SolanaTransactionConfirmationStatus::Finalized,
+                    };
+
+                    Some(RpcConfirmedTransactionStatusWithSignature {
+                        err: err.clone(),
+                        slot: *slot,
+                        memo: None,
+                        block_time: None,
+                        confirmation_status: Some(confirmation_status),
+                        signature: sig.to_string(),
+                    })
+                })
+                .collect()
+        })
+    }
+
+    pub async fn get_signatures_for_address_local_then_remote(
+        &self,
+        client: &SurfnetRemoteClient,
+        pubkey: &Pubkey,
+        config: Option<RpcSignaturesForAddressConfig>,
```

Comment:
We aren't using this config currently for the local. Specifically the `before`, `limit`, and `until` fields need to be used here.

For example, if the limit is 1, and locally we found 1 signature, we don't need to call the remote ctx. 

If locally we have 10 signatures and the limit is 1000, we'll need to truncate the combined list to the limit.

If the use provides the `until` field, and that signature is found locally, no need to go to remote. etc. etc.